src/sql/workbench/contrib/queryHistory/electron-browser/queryHistory.ts

```diff
@@ -57,17 +59,53 @@ export class QueryHistoryWorkbenchContribution implements IWorkbenchContribution
 					// on some events
 					_queryHistoryService.start();
 
-					CommandsRegistry.registerCommand({
-						id: 'queryHistory.clear',
-						handler: (accessor) => {
+					registerAction2(class extends Action2 {
+						constructor() {
+							super({
+								id: `queryHistory.clear`,
+								title: { value: localize('queryHistory.clearLabel', "Clear All History"), original: 'Clear All History' },
+								icon: { id: 'clear-all' },
+								menu: {
+									id: MenuId.ViewTitle,
+									group: 'navigation',
+									when: ContextKeyEqualsExpr.create('view', QUERY_HISTORY_VIEW_ID),
+									order: 1
+								}
+							});
+						}
+
+						run(accessor: ServicesAccessor) {
 							const queryHistoryService = accessor.get(IQueryHistoryService);
 							queryHistoryService.clearQueryHistory();
 						}
 					});
 
-					CommandsRegistry.registerCommand({
-						id: 'queryHistory.toggleCapture',
-						handler: (accessor) => {
+					const PAUSE_QUERY_HISTORY_CAPTURE = localize('queryHistory.disableCapture', "Pause Query History Capture");
+					const START_QUERY_HISTORY_CAPTURE = localize('queryHistory.enableCapture', "Start Query History Capture");
+
+					registerAction2(class extends Action2 {
+						constructor() {
+							super({
+								id: `queryHistory.toggleCapture`,
+								title: { value: START_QUERY_HISTORY_CAPTURE, original: 'Start Query History Capture' },
+								tooltip: START_QUERY_HISTORY_CAPTURE,
+								icon: { id: 'play' },
+								menu: {
+									id: MenuId.ViewTitle,
+									group: 'navigation',
+									when: ContextKeyEqualsExpr.create('view', QUERY_HISTORY_VIEW_ID),
+									order: 1
```

Comment:
Do we want to differentiate the order between 'Clear All History' and  'Start Query History Capture'?

Reply:
Yup, good catch!